File: rest-services/src/main/java/Motociclete/CursaRestClient.java
package Motociclete;

import java.io.*;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Scanner;

public class CursaRestClient {
    private static final String BASE_URL = "http://localhost:8080/motociclete/curse";
    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        while (true) {
            System.out.println("\nChoose an option:\n1. Insert\n2. Update\n3. Delete\n4. List all\n5. Exit");
            String choice = scanner.nextLine();

            switch (choice) {
                case "1" -> insert();
                case "2" -> update();
                case "3" -> delete();
                case "4" -> listAll();
                case "5" -> {
                    System.out.println("Exiting...");
                    return;
                }
                default -> System.out.println("Invalid choice.");
            }
        }
    }

    private static void insert() throws IOException {
        System.out.print("Number of participants: ");
        int numarParticipanti = Integer.parseInt(scanner.nextLine());
        System.out.print("Engine capacity: ");
        int capMotor = Integer.parseInt(scanner.nextLine());

        String json = String.format("{\"numarParticipanti\":%d,\"capMotor\":%d}", numarParticipanti, capMotor);
        HttpURLConnection conn = createConnection(BASE_URL, "POST");
        writeRequest(conn, json);
        int responseCode = conn.getResponseCode();

        if (responseCode == 200 || responseCode == 201) {
            System.out.println("Insert successful!");
            System.out.println(readResponse(conn));
        } else {
            System.out.println("Insert failed with code: " + responseCode);
            System.out.println(readError(conn));
        }
        conn.disconnect();
    }

    private static void update() throws IOException {
        System.out.print("ID to update: ");
        String id = scanner.nextLine();

        System.out.print("New number of participants: ");
        int numarParticipanti = Integer.parseInt(scanner.nextLine());
        System.out.print("New engine capacity: ");
        int capMotor = Integer.parseInt(scanner.nextLine());

        String json = String.format("{\"numarParticipanti\":%d,\"capMotor\":%d}", numarParticipanti, capMotor);
        HttpURLConnection conn = createConnection(BASE_URL + "/" + id, "PUT");
        writeRequest(conn, json);
        int responseCode = conn.getResponseCode();

        if (responseCode == 200) {
            System.out.println("Update successful!");
            System.out.println(readResponse(conn));
        } else {
            System.out.println("Update failed with code: " + responseCode);
            System.out.println(readError(conn));
        }
        conn.disconnect();
    }

    private static void delete() throws IOException {
        System.out.print("ID to delete: ");
        String id = scanner.nextLine();

        HttpURLConnection conn = createConnection(BASE_URL + "/" + id, "DELETE");
        int responseCode = conn.getResponseCode();

        if (responseCode == 200) {
            System.out.println("Delete successful!");
            System.out.println(readResponse(conn));
        } else {
            System.out.println("Delete failed with code: " + responseCode);
            System.out.println(readError(conn));
        }
        conn.disconnect();
    }

    private static void listAll() throws IOException {
        HttpURLConnection conn = createConnection(BASE_URL, "GET");
        int responseCode = conn.getResponseCode();

        if (responseCode == 200) {
            System.out.println("List of Curse:");
            System.out.println(readResponse(conn));
        } else {
            System.out.println("Failed to list curse. Code: " + responseCode);
            System.out.println(readError(conn));
        }
        conn.disconnect();
    }

    private static HttpURLConnection createConnection(String urlString, String method) throws IOException {
        URL url = new URL(urlString);
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod(method);
        conn.setRequestProperty("Content-Type", "application/json");
        conn.setDoOutput(method.equals("POST") || method.equals("PUT"));
        return conn;
    }

    private static void writeRequest(HttpURLConnection conn, String json) throws IOException {
        try (OutputStream os = conn.getOutputStream()) {
            os.write(json.getBytes());
            os.flush();
        }
    }

    private static String readResponse(HttpURLConnection conn) throws IOException {
        try (BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()))) {
            StringBuilder response = new StringBuilder();
            String input;
            while ((input = in.readLine()) != null) {
                response.append(input).append("\n");
            }
            return response.toString();
        }
    }

    private static String readError(HttpURLConnection conn) throws IOException {
        try (BufferedReader in = new BufferedReader(new InputStreamReader(conn.getErrorStream()))) {
            StringBuilder error = new StringBuilder();
            String input;
            while ((input = in.readLine()) != null) {
                error.append(input).append("\n");
            }
            return error.toString();
        }
    }
}

File: CursaRestClient/CursaRestClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CursaRestClient;
class CursaRestClient
{
    private static readonly string BaseUrl = "http://localhost:8080/motociclete/curse";
    private static readonly HttpClient client = new HttpClient();

    static async Task Main()
    {
        while (true)
        {
            Console.WriteLine("\nChoose an option:\n1. Insert\n2. Update\n3. Delete\n4. List all\n5. Exit");
            string choice = Console.ReadLine();

            switch (choice)
            {
                case "1":
                    await Insert();
                    break;
                case "2":
                    await Update();
                    break;
                case "3":
                    await Delete();
                    break;
                case "4":
                    await ListAll();
                    break;
                case "5":
                    Console.WriteLine("Exiting...");
                    return;
                default:
                    Console.WriteLine("Invalid choice.");
                    break;
            }
        }
    }

    private static async Task Insert()
    {
        Console.Write("Number of participants: ");
        int numarParticipanti = int.Parse(Console.ReadLine());
        Console.Write("Engine capacity: ");
        int capMotor = int.Parse(Console.ReadLine());

        var json = $"{{\"numarParticipanti\":{numarParticipanti},\"capMotor\":{capMotor}}}";
        var content = new StringContent(json, Encoding.UTF8, "application/json");

        var response = await client.PostAsync(BaseUrl, content);

        if (response.IsSuccessStatusCode)
        {
            Console.WriteLine("Insert successful!");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
        else
        {
            Console.WriteLine($"Insert failed with code: {response.StatusCode}");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
    }

    private static async Task Update()
    {
        Console.Write("ID to update: ");
        string id = Console.ReadLine();

        Console.Write("New number of participants: ");
        int numarParticipanti = int.Parse(Console.ReadLine());
        Console.Write("New engine capacity: ");
        int capMotor = int.Parse(Console.ReadLine());

        var json = $"{{\"numarParticipanti\":{numarParticipanti},\"capMotor\":{capMotor}}}";
        var content = new StringContent(json, Encoding.UTF8, "application/json");

        var response = await client.PutAsync($"{BaseUrl}/{id}", content);

        if (response.IsSuccessStatusCode)
        {
            Console.WriteLine("Update successful!");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
        else
        {
            Console.WriteLine($"Update failed with code: {response.StatusCode}");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
    }

    private static async Task Delete()
    {
        Console.Write("ID to delete: ");
        string id = Console.ReadLine();

        var response = await client.DeleteAsync($"{BaseUrl}/{id}");

        if (response.IsSuccessStatusCode)
        {
            Console.WriteLine("Delete successful!");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
        else
        {
            Console.WriteLine($"Delete failed with code: {response.StatusCode}");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
    }

    private static async Task ListAll()
    {
        var response = await client.GetAsync(BaseUrl);

        if (response.IsSuccessStatusCode)
        {
            Console.WriteLine("List of Curse:");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
        else
        {
            Console.WriteLine($"Failed to list curse. Code: {response.StatusCode}");
            Console.WriteLine(await response.Content.ReadAsStringAsync());
        }
    }
}
